emcc.py

```diff
@@ -1486,6 +1486,19 @@ def check(input_file):
         if shared.Settings.USE_PTHREADS:
           exit_with_error('LSan currently does not support threads')
 
+      if 'address' in sanitize:
+        shared.Settings.USE_ASAN = 1
+
+        shared.Settings.GLOBAL_BASE = shared.Settings.ASAN_SHADOW_SIZE
+        shared.Settings.TOTAL_MEMORY += shared.Settings.ASAN_SHADOW_SIZE
+        assert shared.Settings.TOTAL_MEMORY < 2**32
+
+        if shared.Settings.SAFE_HEAP:
+          exit_with_error('ASan does not work with SAFE_HEAP')
```

Comment:
is it documented why? seems like it would be redundant but not obvious to me why it would not work.

Reply:
`SAFE_HEAP` instruments shadow memory accesses. Since the shadow memory starts at 0 and we initialize it to some value, that is detected as a null pointer dereference by `SAFE_HEAP`. I'll document this.